core/src/main/java/com/radixdlt/p2p/hostip/NetworkQueryHostIp.java

```diff
@@ -148,33 +147,29 @@ public VotedResult queryNetworkHosts() {
   }
 
   VotedResult get() {
-    return publicIp((count() + 1) / 2); // Round up
-  }
-
-  VotedResult publicIp(int threshold) {
     // Make sure we don't DoS the first one on the list
     Collections.shuffle(this.hosts);
     log.debug("Using hosts {}", this.hosts);
     final Map<HostIp, AtomicInteger> successCounts = Maps.newHashMap();
     final ImmutableMap.Builder<URL, Result<HostIp, IOException>> queryResults =
         ImmutableMap.builder();
+    int maxCount = 0;
+    Optional<HostIp> maxResult = Optional.empty();
     for (URL url : this.hosts) {
       final Result<HostIp, IOException> result = query(url);
       queryResults.put(url, result);
-      final Optional<HostIp> success = result.toOptional();
-      if (success.isPresent()) {
+      if (result.isSuccess()) {
         int newCount =
             successCounts
-                .computeIfAbsent(success.get(), k -> new AtomicInteger())
+                .computeIfAbsent(result.unwrap(), k -> new AtomicInteger())
                 .incrementAndGet();
-        if (newCount >= threshold) {
-          log.debug("Found address {}", success.get());
-          return new VotedResult(success, queryResults.build());
+        if (newCount > maxCount) {
+          maxCount = newCount;
+          maxResult = result.toOptional();
         }
       }
     }
-    log.debug("No suitable address found");
-    return new VotedResult(Optional.empty(), queryResults.build());
```

Comment:
In the case where we have one vote for two different IPs (and the rest of the calls fail), we choose one at random now...

That might be the right move to be defensive and allow the node to boot up... But I think I'd like to handle this a little more clearly. 

In particular:
* If `successCounts.size() == 0` it would be nice to `log.warn` that we couldn't resolve any IP address from external providers - maybe the network is down?
* And if we have `successCounts > 1` then we log with `log.warn`, outputting all the ips and vote counts and that we have arbitrarily resolved `X`?